static/app/components/stories/matrix.tsx

```diff
@@ -1,25 +1,36 @@
-import {type ElementType} from 'react';
+import {ComponentProps, type ElementType} from 'react';
 import {isValidElement} from 'react';
 import styled from '@emotion/styled';
+import first from 'lodash/first';
 
 import SizingWindow from 'sentry/components/stories/sizingWindow';
 import {space} from 'sentry/styles/space';
 
-interface Props {
-  component: ElementType;
-  propMatrix: Record<string, unknown[]>;
-  selectedProps: [string, string];
+export type PropMatrix<E extends ElementType> = Partial<{
+  [Prop in keyof ComponentProps<E>]: Array<ComponentProps<E>[Prop]>;
+}>;
+
+interface Props<E extends ElementType> {
+  propMatrix: PropMatrix<E>;
+  render: ElementType<ComponentProps<E>>;
+  selectedProps: [keyof ComponentProps<E>, keyof ComponentProps<E>];
+  sizingWindowProps?: Partial<ComponentProps<typeof SizingWindow>>;
```

Comment:
this sort of thing explodes typescript if you pass something like a styled('div') because it has 900 props

Reply:
how should we rewrite it?

Reply:
you'd want to export the props directly and avoid using the component in any way. that way you're limited to just the few props in that we actually care about instead of every prop on the forward ref

Reply:
i was mostly talking about the PropMatrix

Reply:
but i guess removing all ComponentProps here would help